mhs/inbound/inbound/request/handler.py

```diff
@@ -51,79 +52,76 @@ def initialize(self, workflows: Dict[str, workflow.CommonWorkflow],
 
         if ref_to_message_id:
-            try:
-                work_description = await wd.get_work_description_from_store(self.work_description_store, ref_to_message_id)
-                logger.info(f'Retrieved work description for message {ref_to_message_id} from state store')
-            except wd.EmptyWorkDescriptionError as e:
-                logger.error(f'No work description found in state store for message {ref_to_message_id}')
-                raise tornado.web.HTTPError(500, f'Unknown message reference {ref_to_message_id}',
-                                            reason="Unknown message reference") from e
+            logger.info(f'RefToMessageId on inbound reply: handling as an referenced reply message')
+            await self._handle_referenced_reply_message(ref_to_message_id, correlation_id, message_data)
         else:
             logger.info(f'No RefToMessageId on inbound reply: handling as an unsolicited message')
-            await self._handle_unsolicited_message(message_id, correlation_id, request_message, received_message)
-            return
+            await self._handle_unsolicited_message(message_id, correlation_id, interaction_id, message_data)
+        self._send_ack(request_message)
 
+    async def _handle_referenced_reply_message(self, message_id: str, correlation_id: str, message_data: MessageData):
+        work_description = await self._get_work_description_from_store(message_id)
         message_workflow = self.workflows[work_description.workflow]
-        logger.info('Forwarding message {message_id} to {workflow}',
-                    fparams={'workflow': message_workflow, 'message_id': ref_to_message_id})
+        logger.info('Forwarding message {message_id} to {workflow}', fparams={'workflow': message_workflow, 'message_id': message_id})
 
         try:
-            await message_workflow.handle_inbound_message(ref_to_message_id, correlation_id, work_description,
-                                                          received_message)
-            self._send_ack(request_message)
+            await message_workflow.handle_inbound_message(message_id, correlation_id, work_description, message_data)
         except Exception as e:
             logger.exception('Exception in workflow')
             raise tornado.web.HTTPError(500, 'Error occurred during message processing, failed to complete workflow',
                                         reason=f'Exception in workflow') from e
 
+    async def _get_work_description_from_store(self, message_id: str):
+        try:
+            work_description = await wd.get_work_description_from_store(self.work_description_store, message_id)
+            logger.info(f'Retrieved work description for message {message_id} from state store')
+            return work_description
+        except wd.EmptyWorkDescriptionError as e:
+            logger.error(f'No work description found in state store for message {message_id}')
```

Comment:
Shouldn't you use `logger.exception()`

Reply:
Not necessarily as the exception here is just to control app flow. I'll change that